js_runner.go

```diff
@@ -66,13 +68,39 @@ func (runner *JSRunner) Init(funcSource string) error {
 		return otto.UndefinedValue()
 	})
 
+	runner.js.Set("console", map[string]interface{}{
+		"error": func(s string) { errorMessageCallback(s)},
+		"log": func(s string) { logMessageCallback(s)},
+	})
+
 	if _, err := runner.SetFunction(funcSource); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+var logMessageCallback = defaultLogFunction
+var errorMessageCallback = defaultLogFunction
+
+var logCallbackLock sync.Mutex
+
+func (runner *JSRunner) SetConsoleErrorCallback(logFn func(string)){
+	logCallbackLock.Lock()
```

Comment:
I'm not sure this locking accomplishes anything.  We're locking while we write the function, but don't have anything blocking attempted read/usage at that time by other goroutines.

To avoid any concurrency headaches, it seems like it would be preferable to add a new version of Init that would accept custom log functions, instead of trying to support customizing logging post-Init.